osutil/sys_linux.go

```diff
@@ -40,3 +40,25 @@ func Symlinkat(target string, newdirfd int, linkpath string) error {
     }
     return nil
 }
+
+// Readlinkat is a direct pass-through to the readlinkat(2) system call.
+func Readlinkat(dirfd int, path string, buf []byte) (n int, err error) {
+    var zero uintptr
+
+    pathPtr, err := syscall.BytePtrFromString(path)
+    if err != nil {
+        return 0, err
+    }
+    var bufPtr unsafe.Pointer
+    if len(buf) > 0 {
+        bufPtr = unsafe.Pointer(&buf[0])
+    } else {
+        bufPtr = unsafe.Pointer(&zero)
```

Comment:
Is this something that's used in Go stdlib too?

Reply:
Yes, I did this by looking at how the generated functions look like.